buildpacks/ruby/src/steps/get_default_process.rs

```diff
@@ -64,16 +64,86 @@ fn detect_web(gem_list: &GemList, app_path: &Path) -> WebProcess {
     }
 }
 
-fn default_rack() -> Process {
-    ProcessBuilder::new(process_type!("web"), "bundle")
-        .args(["exec", "rackup", "--port", "$PORT", "--host", "0.0.0.0"])
+#[derive(Debug, Clone, Eq, PartialEq)]
+enum ShellString {
+    Escape(String),
+    QuoteEnvVar(String),
+}
+
+impl ShellString {
+    fn escape(arg: impl Into<String>) -> ShellString {
+        ShellString::Escape(arg.into())
+    }
+
+    fn quote_env_var(arg: impl Into<String>) -> ShellString {
+        ShellString::QuoteEnvVar(arg.into())
+    }
+}
+
+impl Display for ShellString {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        match self {
+            ShellString::Escape(string) => f.write_str(&shell_words::quote(string)), // single quote only if needed
+            ShellString::QuoteEnvVar(string) => write!(f, "\"{string}\""),
+        }
+    }
+}
+
+fn bashify(program: &ShellString, args: impl IntoIterator<Item = ShellString>) -> Process {
```

Comment:
Food for thought: Do we even need this here? It's only being use twice in the whole code-base and the amount of a) code and b) cognitive overhead it introduces don't seem to be a worthwhile trade-off in my book - especially after we discovered we had to introduce another dependency for it.

I think the code would improve by removing `bashify` (and all supporting code) and just have literals for the two bash commands we have in the buildpack.

Reply:
I think any of the above are fine. Longer term I would ideally want libcnb/libherokubuildpack helper to make this a process a little nicer. 

I ended up reverting the "bashify" stuff and doing this manually. It's (perhaps) more error-prone as escaping must be remembered each time, but is less code and less toil. I honestly don't have a preference for either solution.